### PR TITLE
Upgrade Gradle to 5.0

### PR DESCRIPTION
**Pull request by eskatos**
Mostly by removing the use of the legacy `gradle/project-schema.json` file, [now unsupported](https://github.com/gradle/kotlin-dsl/releases/tag/v1.0.3#breaking-changes), and upgrading AGP from 3.4.0-alpha01 to -alpha04.

A couple minor commits follow the Gradle wrapper upgrade to remove a couple unnecessary bits.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -39,7 +39,7 @@ apply {
     from(file("gradle/compile.gradle.kts"))
 }
 
-android {
+configure<com.android.build.gradle.AppExtension> {
```
Comment (jaredsburrows) [inline build.gradle.kts]: @eskatos We can no longer use `android`?

File: build.gradle.kts [hunk 2]
```diff
@@ -134,6 +134,15 @@ configurations.all {
 }
 
 dependencies {
+
+    val implementation by configurations
```
Comment (jaredsburrows) [inline build.gradle.kts]: @eskatos We have to use delegation for configurations now?

File: build.gradle.kts [hunk 3]
```diff
@@ -182,7 +191,7 @@ dependencies {
     testImplementation(extra["robolectric"] as String)
 }
 
-kapt {
+configure<org.jetbrains.kotlin.gradle.plugin.KaptExtension> {
```
Comment (jaredsburrows) [inline build.gradle.kts]: @eskatos Can't use `kapt` anymore?

File: gradle.properties [hunk 4]
```diff
@@ -1,6 +1,6 @@
 # Gradle
+org.gradle.jvmargs=-Xmx1536m
 org.gradle.caching=true
-org.gradle.configureondemand=true
```
Comment (jaredsburrows) [inline gradle.properties]: Does configure on demand not work or doesnt apply since this project only has 1 module?

File: build.gradle.kts (hunk 1 above)
Comment (eskatos) [reply build.gradle.kts]: Not if you are not using the `plugins {}` block.
See https://docs.gradle.org/current/userguide/kotlin_dsl.html#type-safe-accessors

File: build.gradle.kts (hunk 3 above)
Comment (eskatos) [reply build.gradle.kts]: Not if you are not using the `plugins {}` block.
See https://docs.gradle.org/current/userguide/kotlin_dsl.html#type-safe-accessors

File: gradle.properties (hunk 4 above)
Comment (eskatos) [reply gradle.properties]: It doesn't apply since this project only has 1 module.

**Pull request merged**